tests/test_http.py

```diff
@@ -133,6 +168,28 @@ def test_clients(server):
         c2.close()
 
 
+def test_client_context_with_header_factory(server):
+    with client_context_with_header_factory() as c:
+        assert c.hello("world") == "hello world"
+
+
+def test_client_context_custom_with_header_factory(server):
+    with client_context_with_header_factory() as c:
+        assert c.hello("world") == "hello world"
+
+
+def test_client_with_header_factory(server):
+    c = client_with_header_factory()
+    assert c.hello("world") == "hello world"
+    c.close()
+
+
+def test_client_with_custom_header_factory(server):
+    c = client_with_header_factory()
```

Comment:
I think you want call `client_with_custom_header_factory` here, and this typo caused my previous comment's error not exposed.

Reply:
you are correct, fixed